tests/testsuite/credential_process.rs

```diff
@@ -87,6 +87,31 @@ You may press ctrl-c [..]
         .run();
 }
 
+#[cargo_test]
+fn credential_provider_auth_failure() {
+    let _reg = registry::RegistryBuilder::new()
+        .http_index()
+        .auth_required()
+        .alternative()
+        .no_configure_token()
+        .credential_provider(&["cargo:token-from-stdout", "true"])
+        .build();
+
+    cargo_process("install libc --registry=alternative")
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[UPDATING] `alternative` index
+[ERROR] token rejected for `alternative`
+You may need to log in using this registry's credential provider
+
+Caused by:
+  failed to get successful HTTP response from [..]
+  body:
+  [..]
+"#]])
+        .run();
+}
```

Comment:
Could you add the test in a prior commit, with the test passing?  Then this commit will show how your change affected behavior.